### DISCUSSION TITLE
what is a "generated" column?

### DISCUSSION THREAD

**Discussion by juandent** [fnc12/sqlite_orm#892] [Q&A] [answered]
looking at tests\column_tests.cpp, what is a generated column?

has this something to do with a default value? what is generated_always_as() ? what is the difference between that and just as()?

```C++
auto column = make_column("age", &User::age, generated_always_as(add(&User::id, 5)));
REQUIRE(column.is_generated());
```
thanks!

**Comment by fnc12** [comment, marked as answer]
https://www.sqlite.org/gencol.html

**Reply by juandent** [reply]
Great! and you support virtual and stored!!
